### PR TITLE
any-nix-shell: init at 1.0.0

### PR DESCRIPTION
**Pull request by haslersn**
###### Motivation for this change

any-nix-shell adds fish and zsh support for nix-shell. More supported shells can be added in the future.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/shells/any-nix-shell/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "1fb4c2xkvk3hwy6km02h7j6wsmpachi66n286p2grzgk5k88pijr";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r bin $out
+    wrapProgram $out/bin/any-nix-shell --prefix PATH ":" $out/bin
+    wrapProgram $out/bin/any-nix-shell-wrapper
+    wrapProgram $out/bin/nix-shell-info
```
Comment (veprbl) [inline pkgs/shells/any-nix-shell/default.nix]: Why wrap the last two if you don't specify any action?

Comment (haslersn) [reply pkgs/shells/any-nix-shell/default.nix]: They are bash scripts and wrapping them causes them to find bash.

Comment (veprbl) [reply pkgs/shells/any-nix-shell/default.nix]: I think this happens automatically during ```fixupPhase``` (https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/patch-shebangs.sh) and it has nothing to do with the ```makeWrapper```.

Comment (haslersn) [reply pkgs/shells/any-nix-shell/default.nix]: You are exactly right and I removed the unnecessary wrappers.

**Comment by haslersn** [comment]
This PR is superseded by #46743

**Pull request closed without merging**